Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/vaccination/VaccinationConsentFragmentTest.kt

```diff
@@ -0,0 +1,73 @@
+package de.rki.coronawarnapp.ui.vaccination
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import dagger.Module
+import dagger.android.ContributesAndroidInjector
+import de.rki.coronawarnapp.vaccination.core.VaccinationSettings
+import de.rki.coronawarnapp.vaccination.ui.consent.VaccinationConsentFragment
+import de.rki.coronawarnapp.vaccination.ui.consent.VaccinationConsentViewModel
+import io.mockk.MockKAnnotations
+import io.mockk.impl.annotations.MockK
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import testhelpers.BaseUITest
+import testhelpers.Screenshot
+import testhelpers.SystemUIDemoModeRule
+import testhelpers.TestDispatcherProvider
+import testhelpers.launchFragment2
+import testhelpers.launchFragmentInContainer2
+import testhelpers.takeScreenshot
+import tools.fastlane.screengrab.locale.LocaleTestRule
+
+@RunWith(AndroidJUnit4::class)
+class VaccinationConsentFragmentTest : BaseUITest() {
+    @Rule
+    @JvmField
+    val localeTestRule = LocaleTestRule()
+
+    @get:Rule
+    val systemUIDemoModeRule = SystemUIDemoModeRule()
+
+    @MockK lateinit var vaccinationSettings: VaccinationSettings
+
+    @Before
+    fun setup() {
+        MockKAnnotations.init(this, relaxed = true)
+
+        setupMockViewModel(
+            object : VaccinationConsentViewModel.Factory {
+                override fun create(): VaccinationConsentViewModel =
+                    VaccinationConsentViewModel(
```

Comment:
You can mock the ViewModel itself, this way was required for fastlane

Reply:
updated